From a24e31326de0aef4363fc96e22400cc418274fd9 Mon Sep 17 00:00:00 2001
From: Stanislav Kholmanskikh <stanislav.kholmanskikh@bell-sw.com>
Date: Wed, 12 Aug 2026 10:41:13 +0000
Subject: [PATCH] test/alpine: set LD_LIBRARY_PATH in fakeroot environment

fakeroot prepends `/usr/lib` to LD_LIBRARY_PATH:

  $ LD_LIBRARY_PATH='/some/dir' fakeroot -- sh -c 'echo $LD_LIBRARY_PATH'
  /usr/lib:/usr/lib64/libfakeroot:/usr/lib32/libfakeroot:/some/dir

so the libapk.so.3.0.0 library is taken from the system, and not
the one we just built.

To fix that we set LD_LIBRARY_PATH in the fakeroot environment
prepending it with a dir containing the built libapk.so.3.0.0.
---
 test/alpine/Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/test/alpine/Makefile b/test/alpine/Makefile
index 4f1ea8f0..c73160d8 100644
--- a/test/alpine/Makefile
+++ b/test/alpine/Makefile
@@ -11,7 +11,7 @@ QEMU_EMULATOR	?=
 SUDO		:=
 SYSREPO		:=
 APK		?= $(realpath ../../output/src/apk)
-LD_LIBRARY_PATH	?= ../../output/src
+LD_LIBRARY_PATH	?= $(realpath ../../output/src)
 CBUILD		:= $(arch)
 PWD		:= $(pwd)
 
@@ -40,8 +40,8 @@ repos.stamp: $(repos)
 	if [ -n "$(QEMU_EMULATOR)" ]; then \
 		install -Dm755 "$(QEMU_EMULATOR)" "$(testroot)$(QEMU_EMULATOR)"; \
 	fi; \
-	if ! $(SUDO) LD_LIBRARY_PATH="$(LD_LIBRARY_PATH)" APK="$(APK) --keys-dir /etc/apk/keys" ROOT="$(testroot)" SYSREPO="$(SYSREPO)" \
-		fakeroot sh -ex ./$< > $(basename $@).out 2>&1; then\
+	if ! $(SUDO) APK="$(APK) --keys-dir /etc/apk/keys" ROOT="$(testroot)" SYSREPO="$(SYSREPO)" \
+		fakeroot sh -c 'LD_LIBRARY_PATH="$(LD_LIBRARY_PATH):$${LD_LIBRARY_PATH}" set -ex ./$<' > $(basename $@).out 2>&1; then\
 		echo " FAIL";\
 		printf '%s\n' "------------- output -------------" >&2;\
 		cat $(basename $@).out >&2;\
-- 
2.53.0

